src/form/crd.jl

```diff
@@ -13,8 +13,8 @@ function constraint_pipe_head_loss(
     dhp, dhn = var(wm, n, :dhp_pipe, a), var(wm, n, :dhn_pipe, a)
 
     # Add relaxed constraints for head loss in the positive and negative directions.
-    c_1 = JuMP.@NLconstraint(wm.model, r * head_loss(qp) <= dhp / L)
-    c_2 = JuMP.@NLconstraint(wm.model, r * head_loss(qn) <= dhn / L)
+    c_1 = JuMP.@constraint(wm.model, r * head_loss(wm, qp) <= dhp / L)
```

Comment:
@ccoffrin here's the function tracing in action. We can now explicitly call `head_loss` and have it return an expression that we substitute in.